### PR TITLE
[Serve] Add Multiplex metrics into dashboard

### LINKED ISSUE #37517: [Serve] Multiplex observability improvement
### Description

_No response_

### Use case

_No response_

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?

Number of models per replica:
image
Number of times models loaded
image
Number of times models unloaded
image
Model load latency p99
image
Model unloaded latency p99
image
Registered model
image
Cache hit rate
image
 

## Related issue number

Closes: #37517

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: dashboard/modules/metrics/dashboards/serve_deployment_dashboard_panels.py [hunk 1]
```diff
@@ -149,6 +149,108 @@
         stack=False,
         grid_pos=GridPos(16, 2, 8, 8),
     ),
+    Panel(
```
Comment (edoakes) [inline dashboard/modules/metrics/dashboards/serve_deployment_dashboard_panels.py:152]: for most users these panels will be useless.

Is it possible to put them behind a separate grafana dashboard? or maybe a collapsed-by-default grouping?

**Review by edoakes — APPROVED** [review]
Looks like a great start!

File: dashboard/modules/metrics/dashboards/serve_deployment_dashboard_panels.py (hunk 1 above)
Comment (sihanwang41) [reply dashboard/modules/metrics/dashboards/serve_deployment_dashboard_panels.py:152]: @edoakes I talked to @alanwguo, `row` is not supported in our dashboard, i think I can take it as a follow up to support it. 
issue: https://github.com/ray-project/ray/issues/38024

File: dashboard/modules/metrics/dashboards/serve_deployment_dashboard_panels.py [hunk 2]
```diff
@@ -149,6 +149,108 @@
...
+                legend="{{replica}}",
+            ),
+        ],
+        fill=0,
+        stack=False,
+        grid_pos=GridPos(8, 3, 8, 8),
+    ),
+    Panel(
+        id=12,
+        title="Number of times of multiplexed models unloaded per replica",
```
Comment (alanwguo) [inline dashboard/modules/metrics/dashboards/serve_deployment_dashboard_panels.py:184]:
```suggestion
        title="Multiplexed models unloaded per replica",
```

**Review by alanwguo — COMMENTED** [review]
Mostly copy changes to make graph titles seem more like titles

**Review by alanwguo — APPROVED** [review]

**Comment by sihanwang41** [comment]
@edoakes ping for merge ^^

**Pull request merged**